### PR TITLE
Oracle SOA on Docker

### PR DESCRIPTION
**Pull request by jeqo**
Update the project for Oracle SOA on Docker

### CODE REVIEW

File: OracleSOA/README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+SOA on Docker
+=============
+
+TBD
```
Comment (gvenzl) [inline OracleSOA/README.md]: Please provide a full ReadMe for the build file.

File: OracleSOA/dockerfiles/buildDockerImage.sh [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+#
+# Copyright (c) 2016 TBD
+#
+
+usage() {
+cat << EOF
+
+Usage: buildDockerImage.sh -v [version] [-d | -g | -i] [-s] [-c]
+Builds a Docker Image for Oracle WebLogic.
```
Comment (gvenzl) [inline OracleSOA/dockerfiles/buildDockerImage.sh]: Typo, says WebLogic rather than SOA.

File: OracleSOA/dockerfiles/buildDockerImage.sh [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+cat << EOF
+  SOA Docker Image for '$DISTRIBUTION' version $VERSION is ready to be extended:
+
+    --> $IMAGE_NAME
+
+  Build completed in $BUILD_ELAPSED seconds.
+
+EOF
+else
+  echo "WebLogic Docker Image was NOT successfully created. Check the output and correct any reported problems with the docker build operation."
```
Comment (gvenzl) [inline OracleSOA/dockerfiles/buildDockerImage.sh]: Typo, says WebLogic rather than SOA.

**Comment by jeqo** [comment]
Added.

**Comment by brunoborges** [comment]
@jeqo, this is just a silent install for SOA on a Docker image, but I believe it is not that simple to use it. SOA Suite requires lots of configurations (a database must be running somewhere, for example). Do you have an example on how to actually run it?

**Comment by jeqo** [comment]
@brunoborges yes, I have it working with Ansible playbooks (e.g https://github.com/sysco-middleware/docker-image-oracle-soa-domain and https://github.com/sysco-middleware/docker-compose-oracle-soa-domain), but I'm working in moving those into plain Dockerfiles to upload them as samples. 
You're right, SOA could require lots of configurations, but it also could run in "compact mode" or only run OSB with JavaDB, enough for development and integration tests.

**Comment by jeqo** [comment]
I just add a sample with "compact" OSB domain, no database required.

**Comment by brunoborges** [comment]
Hi @jeqo, since this is quite a big contribution, can you please sign the [OCA](http://www.oracle.com/technetwork/community/oca-486395.html) - Oracle Contributor Agreement? Please cc me in any communication you have through that process (bruno dot borges at oracle dot com).

**Comment by jeqo** [comment]
Done

**Comment by brunoborges** [comment]
Saw you sent OCA. I'll just wait for the OCA committee to confirm they've received and processed. Thanks for proceeding with the paperwork =)

**Comment by brunoborges** [comment]
@jeqo has there been any response from the OCA team?

**Comment by jeqo** [comment]
Nothing yet.

**Comment by klauern** [comment]
I have been testing this branch, and I got an error in running it.  Is there something I can do to increase the disk space for the error given?

```sh
...
[2017-01-10T17:57:34.145+00:00] [as] [NOTIFICATION] [] [oracle.as.install.engine.modules.statistics] [host: e0b3aca5eee2] [nwaddr: 172.17.0.4] [tid: 14] [userId: orac
le] [ecid: 0000La9XeNGFw0H6yvvX6G1OTI1e000004,0] Install Adapter: Mark End for:INSTALL
[2017-01-10T17:57:34.145+00:00] [as] [NOTIFICATION] [] [oracle.as.install.engine.modules.silent] [host: e0b3aca5eee2] [nwaddr: 172.17.0.4] [tid: 14] [userId: oracle]
[ecid: 0000La9XeNGFw0H6yvvX6G1OTI1e000004,0] Processing event:2217
[2017-01-10T17:57:34.145+00:00] [as] [ERROR] [] [oracle.as.install.engine.modules.silent] [host: e0b3aca5eee2] [nwaddr: 172.17.0.4] [tid: 14] [userId: oracle] [ecid:
0000La9XeNGFw0H6yvvX6G1OTI1e000004,0] oracle.sysman.oii.oiif.oiifb.OiifbEndIterateException: java.io.IOException: java.io.IOException: No space left on device[[
oracle.sysman.oii.oiic.OiicInstallAPIException: oracle.sysman.oii.oiif.oiifb.OiifbEndIterateException: java.io.IOException: java.io.IOException: No space left on devi
ce
        at oracle.sysman.oii.oiic.OiicAPIInstaller.doOperation(OiicAPIInstaller.java:1004)
        at oracle.as.install.engine.modules.install.action.NextGenOUIInstaller.start(NextGenOUIInstaller.java:391)
        at oracle.as.install.engine.modules.install.action.NextGenInstallManager.launchOUI(NextGenInstallManager.java:158)
        at oracle.as.install.engine.modules.install.InstallModule.launchOUI(InstallModule.java:164)
        at oracle.as.install.engine.modules.install.InstallModule$1.run(InstallModule.java:261)
Caused by: oracle.sysman.oii.oiif.oiifb.OiifbEndIterateException: oracle.sysman.oii.oiif.oiifb.OiifbEndIterateException: java.io.IOException: java.io.IOException: No
space left on device
        at oracle.sysman.oii.oiic.OiicInstallAPISession.doOperation(OiicInstallAPISession.java:500)
        at oracle.sysman.oii.oiic.OiicAPIInstaller.doOperation(OiicAPIInstaller.java:971)
        ... 4 more
Caused by: oracle.sysman.oii.oiif.oiifb.OiifbEndIterateException: java.io.IOException: java.io.IOException: No space left on device
        at oracle.sysman.oii.oiij.OiijNextGenCopyPhase.copyNextGenCompFiles(OiijNextGenCopyPhase.java:145)
        at oracle.sysman.oii.oiif.oiifw.OiifwInstallPhaseWCCE$OiifwComponentIP1WCCE.doOperation(OiifwInstallPhaseWCCE.java:2115)
        at oracle.sysman.oii.oiif.oiifb.OiifbCondIterator.iterate(OiifbCondIterator.java:171)
        at oracle.sysman.oii.oiif.oiifw.OiifwInstallPhaseWCCE.doOperation(OiifwInstallPhaseWCCE.java:1023)
        at oracle.sysman.oii.oiif.oiifb.OiifbCondIterator.iterate(OiifbCondIterator.java:171)
        at oracle.sysman.oii.oiif.oiifw.OiifwActionsPhaseWCDE.doOperation(OiifwActionsPhaseWCDE.java:699)
        at oracle.sysman.oii.oiif.oiifb.OiifbLinearIterator.iterate(OiifbLinearIterator.java:147)
        at oracle.sysman.oii.oiic.OiicNextGenInstallAPISession$OiicNextGenAPISelCompsInstall.doOperation(OiicNextGenInstallAPISession.java:434)
        at oracle.sysman.oii.oiif.oiifb.OiifbCondIterator.iterate(OiifbCondIterator.java:171)
        at oracle.sysman.oii.oiic.OiicInstallAPISession.doInstallAction(OiicInstallAPISession.java:689)
        at oracle.sysman.oii.oiic.OiicInstallAPISession.access$000(OiicInstallAPISession.java:91)
        at oracle.sysman.oii.oiic.OiicInstallAPISession$OiicActionsThread.run(OiicInstallAPISession.java:973)
Caused by: java.io.IOException: java.io.IOException: No space left on device
        at oracle.sysman.oii.oiij.OiijNextGenJarExtracter.copyJarContents(OiijNextGenJarExtracter.java:341)
        at oracle.sysman.oii.oiij.OiijNextGenJarExtracter.extract(OiijNextGenJarExtracter.java:421)
        at oracle.sysman.oii.oiij.OiijNextGenCopyPhase.extractJar(OiijNextGenCopyPhase.java:208)
        at oracle.sysman.oii.oiij.OiijNextGenCopyPhase.processComponentJars(OiijNextGenCopyPhase.java:167)
        at oracle.sysman.oii.oiij.OiijNextGenCopyPhase.copyNextGenCompFiles(OiijNextGenCopyPhase.java:136)
        ... 11 more
Caused by: java.io.IOException: No space left on device
        at java.io.FileOutputStream.writeBytes(Native Method)
        at java.io.FileOutputStream.write(FileOutputStream.java:345)
        at java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:82)
        at java.io.BufferedOutputStream.write(BufferedOutputStream.java:126)
        at oracle.sysman.oii.oiix.OiixFileOps.copyStream(OiixFileOps.java:1553)
        at oracle.sysman.oii.oiij.OiijFastJarExtracter.copyFileFromJar(OiijFastJarExtracter.java:337)
        at oracle.sysman.oii.oiij.OiijNextGenJarExtracter.copyFileFromJar(OiijNextGenJarExtracter.java:433)
        at oracle.sysman.oii.oiij.OiijFastJarExtracter.extractEntry(OiijFastJarExtracter.java:272)
        at oracle.sysman.oii.oiij.OiijNextGenJarExtracter.copyJarContents(OiijNextGenJarExtracter.java:229)
        ... 15 more

]]
[2017-01-10T17:57:34.146+00:00] [sysman] [NOTIFICATION] [] [oracle.sysman.oii.oiio.oiiol.OiiolTextLogger] [host: e0b3aca5eee2] [nwaddr: 172.1[root@e0b3aca5eee2 OraIns
tall2017-01-10_05-56-21PM]# exit
```

**Comment by klauern** [comment]
I broke out the `RUN .. && \ ... \` command into separate `RUN` commands, and opened up the one right before kicking off the installer in a bash prompt.

```
12:00 $ docker run -it f9062d3ff4f1 /bin/bash
[root@63de6548649e /]# df -kh
Filesystem                                                                                       Size  Used Avail Use% Mounted on
/dev/mapper/docker-8:2-1572919-9c2d2572ce7638908b153d826e25276f909135fba3f8d4940e2866e6dd5474f7   10G  6.5G  3.6G  65% /
tmpfs                                                                                            3.9G     0  3.9G   0% /dev
tmpfs                                                                                            3.9G     0  3.9G   0% /sys/fs/cgroup
/dev/sda2                                                                                        226G  126G   89G  59% /etc/hosts
shm                                                                                               64M     0   64M   0% /dev/shm
```

It appears that right before running the installer, we're already at 65% use on the `/` mount point.  I don't quite know how to expand this, however, and my own machine has well over 100GB free to use, so I know the disk space issue isn't coming from the host itself, but within the container.

**Comment by Djelibeybi** [comment]
@klauern this is one of the problems if you're using the devicemapper storage driver, as it creates a 10GB loop-mount device by default. I strongly recommend switching to either `btrfs` or `overlayfs` as your storage driver. 

If you can't do that, you'll need to increase the size of the devicemapper loop device that Docker creates for each container. See https://docs.docker.com/engine/userguide/storagedriver/device-mapper-driver/ for more info on that.

**Comment by klauern** [comment]
@Djelibeybi thanks for the pointer.  I am still new to Docker, so I was unaware of this limitation.  I'll look at how I can switch the devicemapper or increase the size of the current one.  Thank you for the information!

**Comment by Djelibeybi** [comment]
Closing because we do not support JDK7. If this can be rebased to use the existing Server JRE 8 image, please feel free to resubmit a new Dockerfile.

**Comment by jeqo** [comment]
Didn't know that JDK7 is not supported, the image still in the repo though. There is any place where I can find that info?
I cannot do a rebase to JRE 8 because SOA 12.1.3 does not support Java 8.  
Also more than 4 months passed and I haven't received any notice about the OCA.

**Comment by Djelibeybi** [comment]
Assigning across to @brunoborges who can follow-up on the OCA. Is there a version of SOA that does support JDK8? If so, we should probably use that instead.

**Comment by jeqo** [comment]
SOA 12.2.1 should work with JDK 8. I will migrate it. Looking forward to the OCA.

**Comment by Djelibeybi** [comment]
That's great, thanks. @brunoborges should be able to follow up on the OCA.

**Comment by jeqo** [comment]
Could this PR be reopened? or should I create a new one?

**Comment by Djelibeybi** [comment]
Please submit a new PR.

**Comment by Djelibeybi** [comment]
I'm not reopening this PR, but I just wanted to let @jeqo know that I want to apologise for not getting your CLA processed quickly and efficiently and as such, I've taken over processing CLAs for this repository and have sent you a request to resubmit your CLA so I can process it.

**Pull request closed without merging**